File: java/vacuumba.java
import java.util.Scanner;

public class vacuumba {
	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		int n = scan.nextInt();
		while (n-- > 0) {
			int m = scan.nextInt();
			double X = 0;
			double Y = 0;
			double direction = 90;
			while (m-- > 0) {
				double angle = scan.nextDouble();
				double distance = scan.nextDouble();
				direction += angle;
				double direction_radian = direction * Math.PI / 180;
				double radian_cos = Math.cos(direction_radian);
				double radian_sin = Math.sin(direction_radian);
                if (direction == 90)
                {
                    radian_cos = 0;
                    radian_sin = 1;
                }
                else if(direction == -90)
                {
                    radian_cos = 0;
                    radian_sin = -1;
                }
				X += distance * radian_cos;
				Y += distance * radian_sin;
			}
			System.out.println(X + " " + Y);
		}
		scan.close();
	}
}

File: ts/src/vacuumba.ts
import readline = require('readline')

interface Direction {
  angle: number
  distance: number
}

interface Point {
  x: number
  y: number
}

function getLocation (segments: Direction[]): Point {
  const point = { x: 0, y: 0 }
  let dir = 90
  segments.forEach(segment => {
    dir += segment.angle
    const radianDir = getRadian(dir)

    point.x += segment.distance * Math.cos(radianDir)
    point.y += segment.distance * Math.sin(radianDir)
  })

  return point
}

function getRadian (degrees: number): number {
  return degrees * Math.PI / 180
}

function printLocation (segments: Direction[]): void {
  if (segments.length > 0) {
    const location = getLocation(segments)
    console.log(`${location.x.toFixed(6)} ${location.y.toFixed(6)}`)
  }
}

const digitsOnly = /^\d+$/
let segments: Direction[] = []
let lineCount = 0

readline
  .createInterface({
    input: process.stdin,
    output: process.stdout
  })
  .on('line', (line: string) => {
    if (lineCount > 0) {
      if (digitsOnly.test(line)) {
        printLocation(segments)
        segments = []
      } else {
        const [angle, distance] = line.split(' ').map(Number)
        segments.push({ angle: angle ?? 0, distance: distance ?? 0 })
      }
    }
    lineCount++
  })
  .on('close', () => {
    printLocation(segments)
    process.exit(0)
  })
